src/sql/src/plan/query.rs

```diff
@@ -214,26 +214,30 @@ pub fn plan_insert_query(
         InsertSource::DefaultValues => unsupported!("INSERT ... DEFAULT VALUES"),
     };
 
-    // Validate that the arity of the source query matches the arity of the
-    // target table.
     let typ = qcx.relation_type(&expr);
-    if typ.arity() != desc.arity() {
-        bail!(
-            "INSERT statement specifies {} columns, but table has {} columns",
-            typ.arity(),
-            desc.arity(),
-        );
+
+    // Validate that the arity of the source query is at most the size of declared columns or the
```

Comment:
I see that `source` was brought over from a previous commit! But same thought here.